exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt

```diff
@@ -76,50 +76,159 @@ class Database private constructor(private val resolvedVendor: String? = null, v
             dialects[prefix] = dialect
         }
 
+        /**
+         * Attempts to make a connection to a database.
+         *
+         * @param explicitVendor    the vendor name, if it's `null` then it will be resolved
+         * from database dialect name
+         * @param getNewConnection  used to create a new connection
+         * @param schema            the default schema to use
+         * @param setupConnection   used to apply operations on the established connection
+         * @param manager           used to register a transaction manager
+         */
         private fun doConnect(
             explicitVendor: String?,
             getNewConnection: () -> Connection,
+            schema: Schema? = null,
             setupConnection: (Connection) -> Unit = {},
             manager: (Database) -> TransactionManager = { ThreadLocalTransactionManager(it, DEFAULT_REPETITION_ATTEMPTS) }
         ): Database {
-            return Database(explicitVendor) {
+            val database = Database(explicitVendor) {
                 connectionInstanceImpl(getNewConnection().apply { setupConnection(this) })
             }.apply {
                 TransactionManager.registerManager(this, manager(this))
             }
+
+            if(schema != null) {
+                with(database.transactionManager.newTransaction()) {
```

Comment:
I guess it should be replaced with `transaction` function or create transaction will not be closed ever.

Reply:
I think Database.connect(schema = Schema("myschema")) could be calling outside a `transaction` and all next transactions should have the passed `schema` as the default schema. What do you think?

Reply:
I miss it but looks like `setScheme` will be called only once on some connection and all other connections will remain with their default scheme.

I think you have to call setScheme just after a new transaction was started.

Reply:
I did some changes but I'm not sure if this is a correct behavior

````KOTLIN
connect(schema = myschema)

transaction {
   // Default schema : myschema
}
````

// OR :

````KOTLIN

connect()
transaction { 
// Default schema : PUBLIC
 }

transaction {
  connect(schema = myschema)
  transaction {
    // Default schema : myschema
  }
  // Default schema : myschema
}
transaction {
   // Default schema : myschema
}
````

Reply:
@Tapac as you mentioned `setSchema` will change schema for some connection. So to do that you can simply call `SchemaUtils.setSchema()` inside a transaction. So I think the extension in this PR becomes useless. Do you agree?